25-multiplecomponents-lab_v16/src/03-shoppinglist-filters.jsx

```diff
@@ -1,7 +1,144 @@
+            height: "25px",
+            letterSpacing: 0,
+            overflow: "hidden",
+            fontSize: 14,
+        }
+
+        let itemsList = this.state.items
+        let filterTxt = this.state.filterText
+
+        const filterByName = () => {
+            let filter = this.nameFilter.value
+                this.setState({ 
+                    filterText: filter
+                });
+        }
+
+      
+        if(filterTxt.length>0){
+
+            let ischecked = this.boughtItems.checked
+        
+            itemsList =   itemsList.filter(element => element.name === filterTxt);
+         }  
+
+        return (
+            <div>
+                 <table>
+                    <tbody>
+                    <tr>
+                            <td>Name:</td>
+                            <td><input type="text" ref={name => { this.name = name }} /></td>
+                    </tr>
+                    <tr>
+                            <td>Image:</td>
+                            <td> <input type="text"  ref={url => { this.url = url }}/> Use full path</td>
+                    </tr>
+                    <tr>
+                            <td>Unit:</td>
+                            <td><input type="text"  ref={unit => { this.unit = unit }} /></td>
+                    </tr>
+                    <tr>
+                            <td></td>
+                            <td><input type="button" value="Add Item" onClick={this.add} /><p /></td>
+                    </tr>
+                  
+                
+                   
+                  </tbody>
+               </table>
+              
+                <table style={style}> 
+                <tbody>
+                     <tr>
+                        <td></td>
+                        <td></td>
+                        <td></td>
+                        <td></td>
+                        <td></td>
+                    </tr>
+                    <tr>
+                        <td>Item Name <input type="text" onChange={() =>  filterByName(itemsList)} ref={nameFilter => { this.nameFilter = nameFilter }} /></td>
+                        <td>Item Image</td>
+                        <td>Item Unit </td>
+                        <td></td>
+                        <td></td>
+                    </tr>
+                    {itemsList.map((element, index) => {
+                        return <tr key={index}>
+                            <td>{element.name}</td>
+                            <td><img src={element.url}  height="42" width="42"/></td>
+                            <td>{element.unit}</td>
+                            <td>
+                            <input type="text"  ref={unitsNo => { this.unitsNo = unitsNo }} />
+                            <input  type="button" value="Bye!"  onClick={() => this.buy(element.id)} />
+                            </td>
+                            <td><input type="button" value="Delete" onClick={() => this.delete(index)} /></td>

```

Comment:
Note that since you call the functions with an explicit `this` (such as: `() => this.delete(index)`, you don't need to use the arrow function assignment in the class definition, and can use a normal function